modules/issue_tracker/ajax/EditIssue.php

```diff
@@ -0,0 +1,619 @@
+<?php
+/**
+ * Issue tracker
+ *
+ * Handles issue edits and returns data in response to a front end call.
+ *
+ * PHP Version 5
+ *
+ * @category Loris
+ * @package  Media
+ * @author   Caitrin Armstrong <caitrin.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris-Trunk
+ */
+
+/**
+ * Issue tracker
+ *
+ * Handles issue edits and returns data in response to a front end call.
+ *
+ * PHP Version 5
+ *
+ * @category Loris
+ * @package  Media
+ * @author   Caitrin Armstrong <caitrin.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris-Trunk
+ */
+
+require_once "Email.class.inc";
+
+//TODO: or split it into two files... :P
+if (isset($_GET['action'])) {
+    $action = $_GET['action'];
+    if ($action == "getData") {
+        echo json_encode(getIssueFields());
+    } else if ($action == "edit") {
+        echo json_encode(editIssue());
+    }
+}
+
+//TODO: encapsulate more
+/**
+ * Either updates an old issue or adds a new one
+ *
+ * @throws DatabaseException
+ *
+ * @return array indicating insertion success or failure
+ */
+function editIssue()
+{
+    $db =& Database::singleton();
+    $user =& User::singleton();
+
+    $issueValues = array();
+    $validateValues = array();
+    $fields = array(
+        'assignee',
+        'status',
+        'priority',
+        'visitLabel',
+        'centerID',
+        'title',
+        'category',
+        'module',
+        'watching',
+    );
+    $fieldsToValidateFirst = array(
+        'PSCID',
+        'visitLabel',
+    );
+
+    foreach ($fields as $field) {
+        if (isset($_POST[$field])) {
+            $issueValues[$field] = $_POST[$field];
+        }
+    }
+
+    foreach ($fieldsToValidateFirst as $vField) {
+        if (isset($_POST[$vField])) {
+            $validateValues[$vField] = $_POST[$vField];
+        }
+    }
+
+    $issueID = $_POST['issueID'];
+    $issueValues['lastUpdatedBy'] = $user->getData('UserID');
+
+    //this whole validation thing really needs some best practice love
+    $validatedInput = validateInput($validateValues, $issueID);
+    if ($validatedInput['isValidSubmission']) {
+        if (array_key_exists('sessionID', $validatedInput)) {
+            $issueValues['sessionID'] = $validatedInput['sessionID'];
+        }
+        if (array_key_exists('candID', $validatedInput)) {
+            $issueValues['candID'] = $validatedInput['candID'];
+        }
+    } else {
+        return $validatedInput;
+    } //aka when it's not valid
+
+    if (!empty($issueID)) {
```

Comment:
When you receive a null value from react is can sometimes be in the string format "null", which seems to be the case now. So `if (!empty("null"))` returns true, which is probably not the desired behaviour. [See Docs](http://php.net/manual/en/function.empty.php)